transcom/mymove
# B 21455 int

## [B-21455](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1015558)

## Summary

This change update the service counselor and TOO roles to be able update ppm shipments that have a estimated incentive of zero. An estimated incentive of zero can, currently, only be achieved when a customer pushes a PPM when DTOD is down. 

The Service Counselor role updates the ppms when they select the 'Submit move details' button.

The TOO updates the ppms when the attempt to approve either the move or an individual shipment.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Service Counselor:
1. Set DTOD_SIMULATE_OUTAGE to true and run the app
2. Access the Customer application
3. Login as a Customer
4. Create a Move
5. Add a PPM
6. Submit the move for review
7. Validate the the estimated incentive is zero in the ppm_shipments table for the shipment you created
8. Set DTOD_SIMULATE_OUTAGE to false and re-rerun the server (you can also run the test without this step, but you will be unable to approve the move and will instead see a message about DTOD being down)
9. Login as a Service Counselor
10. Select the move you created
11. Fill in missing details 
12. Submit the move
13. Validate the the estimated incentive in the ppm_shipments table is no longer zero for the shipment you created

##THE ABILITY FOR TOO TO APPROVE PPMS IS CURRENTLY BROKEN IN INTEGRATION DUE TO ANOTHER TASK
TOO:
1. Set DTOD_SIMULATE_OUTAGE to true and run the app
2. Access the Customer application
3. Login as a Customer
4. Create a Move with the Current Duty Station set to 'NS Mayport' (this allows the TOO to approve ppm shipments without the SC approval)
5. Add a PPM
6. Submit the move for review
7. Validate the the estimated incentive is zero in the ppm_shipments table for the shipment you created
10. Set DTOD_SIMULATE_OUTAGE to false and re-rerun the server (you can also run the test without this step, but you will be unable to approve the move and will instead see a message about DTOD being down)
11. Login as a TOO
12. Select the move you created
13. Fill in missing details 
14. Select the PPM shipment for approval
15. Select the Counseling check box
16. Attempt to approve the Move and ppm shipment you created
17. Validate the the estimated incentive in the ppm_shipments table is no longer zero for the shipment you created


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
